docker/k8s-spark-operator/Dockerfile

```diff
@@ -0,0 +1,71 @@
+RUN mkdir /opt/hadoop-${HADOOP_VERSION}/logs
+
+ENV HADOOP_PREFIX=/opt/hadoop-${HADOOP_VERSION}
+ENV HADOOP_CONF_DIR=/etc/hadoop
+
+#Hive
+ENV HIVE_HOME=/opt/hive
+ENV HIVE_VERSION=2.3.3
+RUN wget -q https://archive.apache.org/dist/hive/hive-$HIVE_VERSION/apache-hive-$HIVE_VERSION-bin.tar.gz \
+	&& tar -xzf apache-hive-$HIVE_VERSION-bin.tar.gz \
+	&& mv apache-hive-$HIVE_VERSION-bin $HIVE_HOME \
+	&& rm apache-hive-$HIVE_VERSION-bin.tar.gz
+
+#Hudi
+ENV HUDI_VERSION=0.5.3
+RUN wget -q https://repo1.maven.org/maven2/org/apache/hudi/hudi-hive-bundle/$HUDI_VERSION/hudi-hive-bundle-$HUDI_VERSION.jar \
+    && mv hudi-hive-bundle-$HUDI_VERSION.jar $HIVE_HOME/lib
+RUN wget -q https://repo1.maven.org/maven2/org/apache/hudi/hudi-hadoop-mr-bundle/$HUDI_VERSION/hudi-hadoop-mr-bundle-$HUDI_VERSION.jar \
+    && mv hudi-hadoop-mr-bundle-$HUDI_VERSION.jar $HIVE_HOME/lib
+
+#AWS
+ARG AWS_SDK_VERSION=1.11.699
+ARG HADOOP_AWS_VERSION=3.0.3
+ARG HTTPCLIENT_VERSION=4.5.11
+
+RUN wget -q https://repo1.maven.org/maven2/net/logstash/log4j/jsonevent-layout/1.7/jsonevent-layout-1.7.jar -P $SPARK_HOME/jars/
+RUN wget -q https://repo1.maven.org/maven2/net/minidev/json-smart/1.1.1/json-smart-1.1.1.jar -P $SPARK_HOME/jars/
+RUN wget -q https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-aws/${HADOOP_AWS_VERSION}/hadoop-aws-${HADOOP_AWS_VERSION}.jar -P $SPARK_HOME/jars/
+RUN wget -q https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk/${AWS_SDK_VERSION}/aws-java-sdk-${AWS_SDK_VERSION}.jar -P $SPARK_HOME/jars/
+RUN wget -q https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-core/${AWS_SDK_VERSION}/aws-java-sdk-core-${AWS_SDK_VERSION}.jar -P $SPARK_HOME/jars/
+RUN wget -q https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-s3/${AWS_SDK_VERSION}/aws-java-sdk-s3-${AWS_SDK_VERSION}.jar -P $SPARK_HOME/jars/
+RUN rm -f $SPARK_HOME/jars/httpclient-*.jar && wget -q https://repo1.maven.org/maven2/org/apache/httpcomponents/httpclient/${HTTPCLIENT_VERSION}/httpclient-${HTTPCLIENT_VERSION}.jar -P /spark/jars
+
+#Python
+RUN apt-get update \
+    && apt-get install -y coreutils jq less inotify-tools python3 python3-setuptools \
+    && curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py \
+    && python3 get-pip.py 'pip==20.1.1' \
+    && rm get-pip.py \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY conf/log4j.json.properties /opt/spark/custom/conf/
+ENV PYTHONHASHSEED 1
+ENV PATH="${HADOOP_PREFIX}/bin:${SPARK_HOME}/bin:${PATH}"
+ENV SPARK_DIST_CLASSPATH /etc/hadoop:/opt/hadoop-2.9.2/share/hadoop/common/lib/*:/opt/hadoop-2.9.2/share/hadoop/common/*:/opt/hadoop-2.9.2/share/hadoop/hdfs:/opt/hadoop-2.9.2/share/hadoop/hdfs/lib/*:/opt/hadoop-2.9.2/share/hadoop/hdfs/*:/opt/hadoop-2.9.2/share/hadoop/yarn:/opt/hadoop-2.9.2/share/hadoop/yarn/lib/*:/opt/hadoop-2.9.2/share/hadoop/yarn/*:/opt/hadoop-2.9.2/share/hadoop/mapreduce/lib/*:/opt/hadoop-2.9.2/share/hadoop/mapreduce/*:/contrib/capacity-scheduler/*.jar
```

Comment:
What's this?

Reply:
@lyogev we need it to let spark know where the new hadoop is sitting, in the old docker we used `export SPARK_DIST_CLASSPATH=$(hadoop classpath)` in the spark-env.sh. since we dont use it anymore we need to hard code it here. 
Also im using `SPARK_EXTRA_CLASSPATH`, if i'm not setting this the workers are not familiar with the new hadoop installation.